dockstore-integration-testing/src/test/java/io/dockstore/client/cli/ExtendedTRSOpenApiIT.java

```diff
@@ -72,27 +98,227 @@
 
     @SystemStub
     public final SystemErr systemErr = new SystemErr();
 
+    @BeforeAll
+    public static void setup() throws Exception {
+        bucketName = SUPPORT.getConfiguration().getMetricsConfig().getS3BucketName();
+        s3EndpointOverride = SUPPORT.getConfiguration().getMetricsConfig().getS3EndpointOverride();
+        metricsDataClient = new MetricsDataS3Client(bucketName, s3EndpointOverride);
+        // Create a bucket to be used for tests
+        s3Client = TestUtils.getClientS3V2(); // Use localstack S3Client
+        createBucket(s3Client, bucketName);
+        deleteBucketContents(s3Client, bucketName); // This is here just in case a test was stopped before tearDown could clean up the bucket
+    }
+
     @BeforeEach
     @Override
     public void resetDBBetweenTests() throws Exception {
         CommonTestUtilities.cleanStatePrivate2(SUPPORT, false, testingPostgres);
     }
 
+    @AfterEach
+    public void tearDown() {
+        // Delete all objects from the S3 bucket after each test
+        deleteBucketContents(s3Client, bucketName);
+    }
+
+    /**
+     * Test submitting metrics data using the Extended GA4GH endpoint.
+     *
+     * At the end of this test, the S3 folder structure should look like the following. Note that OBJECT METADATA is the S3 object metadata and is not part of the folder structure
+     * local-dockstore-metrics-data
+     * ├── tool
+     * │   └── quay.io
+     * │       └── dockstoretestuser2
+     * │           └── dockstore-cgpmap
+     * │               └── symbolic.v1
+     * │                   └── TERRA
+     * │                       └── 1673972062578.json
+     * │                           └── OBJECT METADATA
+     * │                               └── owner: 1
+     * │                               └── description:
+     * └── workflow
+     *     └── github.com
+     *         └── DockstoreTestUser2
+     *             └── dockstore_workflow_cnv%2Fmy-workflow
+     *                 └── master
+     *                     ├── TERRA
+     *                     │   └── 1673972062578.json
+     *                     │       └── OBJECT METADATA
+     *                     │           └── owner: 1
+     *                     │           └── description: A single execution
+     *                     └── DNA_STACK
+     *                         └── 1673972062578.json
+     *                             └── OBJECT METADATA
+     *                                 └── owner: 1
+     *                                 └── description: A single execution
+     * @throws IOException
+     */
+    @Test
+    void testSubmitMetricsData() throws IOException {
+        // Admin user
+        final ApiClient webClient = getOpenAPIWebClient(USER_2_USERNAME, testingPostgres);
+        final WorkflowsApi workflowApi = new WorkflowsApi(webClient);
+        final io.dockstore.openapi.client.api.UsersApi usersApi = new io.dockstore.openapi.client.api.UsersApi(webClient);
```

Comment:
Nit: if this is a class for testing only OpenApi, these package qualifiers should be unnecessary